src/lib/comms/include/sol-http-server.h

```diff
@@ -208,6 +218,50 @@ int sol_http_server_set_last_modified(struct sol_http_server *server, const char
 int sol_http_server_send_response(struct sol_http_request *request, struct sol_http_response *response);
 
 /**
+ * @brief Send the response and keep connection alive to request given in the
+ * callback registered on @c sol_http_server_register_handler.
+ *
+ * After this call, the caller will be resposible to close the connection calling
+ * @c sol_http_response_sse_del. @note all the necessary headers are set by this function.
+ *
+ * @param request The request given on the callback of
+ * @c sol_http_server_register_handler.
+ * @param cb Callback called when the connection is closed from the client side.
+ * @param cb_data The data pointer to be passed to @a cb.
+ *
+ * @return @c sol_http_response_sse on success, @c NULL otherwise.
+ *
+ * @see sol_http_response_sse_del
+ * @see sol_http_response_sse_send_data
+ */
+struct sol_http_response_sse *sol_http_server_send_response_sse(struct sol_http_request *request,
```

Comment:
as above I'd not call it "see", but rather take a `sol_http_response` and return a progressive handle to it. Like:
```c
struct sol_http_response_progressive *sol_http_server_send_progressive_response(struct sol_http_request *request, struct sol_http_response *response, void (*on_close)(void *data.. ));
```

This would not set any SSE specific header, these must be done by the user. You can offer a helper to set the standard http headers used by SSE applications as a separate function (static inline in the header)

Extra comments:
 - why is `response` non-const in send_response?
 - rename `cb` to `on_close`.